packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
       }
     } else if ("value" in desc && desc.writable) {
       desc.value = value;
```

Comment:
IMHO `DefineOwnProperty` is not the same with `receiver[property] = value`. 
Think this example.

```
class A {
}

class B extends A {
    constructor() {
        super()
        super.x = 3
        console.log(this.x)
        console.log(super.x)
    }

    set x(v) {
        console.log('i am invoked!') // will not invoked
        this._x = v
    }
    get x() {
        return this._x
    }
}
new B()
```